### ISSUE TITLE
Can't use find with spaces / consecutive words.

### ISSUE THREAD

**Issue by jolmg** [djcb/mu#1128]
## Expected or desired behavior

I expected,
```
mu find subject:"some string"
```
to match emails that included the string "some string".

## Actual behavior

Instead, I get,
```
mu: no matches for search expression (4)
```

This also happens with other selectors like `body` and `from`.

I can match the emails I want with:
```
mu find subject:"some*string"
```
but that also matches "some other string".

## Steps to reproduce

1. Chose an email that has consecutive words in `from` or `subject`.
2. Try to match it with `mu find subject:"your subject"`

## Versions of mu, mu4e/emacs, operating system etc.

mu: 0.9.18
mu4e: from the same release of mu 0.9.18
emacs: 25.2.1
operating system: Archlinux

## Any other detail

I'm not sure if this should be a separate issue, but I also expected,
```
$ mu find subject:"some" subject:"string"
mu: no matches for search expression (4)
```
to match.

**Comment by jolmg** [comment]
Thank you for your work, by the way. Let me know if I can help you debug this.

**Comment by djcb** [comment]
mu4e uses the Xapian query language, which is a bit limited with respect to anything that is not a normal word. mu4e preprocess a bit to work around this... but it's far from perfect as you've noticed!

The real fix would be to write a mu-specific query-parser.... which will happen at some point.

**Comment by wavexx** [comment]
On Fri, Sep 22 2017, Dirk-Jan C. Binnema wrote:
> mu4e uses the Xapian query language, which is a bit limited with
> respect to anything that is not a normal word. mu4e preprocess a bit
> to work around this... but it's far from perfect as you've noticed!

Xapian does seem to support phrase search though.

https://xapian.org/docs/queryparser.html

Any reason it's not working as it is?
It seems to work fine in other tools using xapian (recoll, or
axi-cache on debian).

I was even surprised by re-reading the documentation it does
distribution over (), which is pretty sweet.

**Comment by djcb** [comment]
We have a new custom query-parser now (git); check the `mu-query` database for some details. 
Note that  `mu find subject:"your subject"` can't really work 'as expected', since the shell will eat the quotes.

Anyway, with the new parser, currently the easiest way would be to use a regexp, ie.
`mu find subject:/your.subject/`
(spaces are troublesome still).

**Comment by wavexx** [comment]
On Wed, Oct 25 2017, Dirk-Jan C. Binnema wrote:
> Closed #1128.

I didn't have a look at the new parser yet, but from the top of my mind,
would by chance this fix #999 as well?
